internal/stream_fetch_helper.go

```diff
@@ -67,46 +81,76 @@ func DownloadAndDecompressStream(backup *Backup) error {
 	return NewArchiveNonExistenceError(fmt.Sprintf("Archive '%s' does not exist.\n", backup.Name))
 }
 
-// GetOperationLogsCoveringInterval lists the operation logs that cover the interval
-func GetOperationLogsCoveringInterval(folder storage.Folder, start time.Time, end *time.Time) ([]storage.Object, error) {
-	oplogFiles, _, err := folder.ListFolder()
+// GetLogsCoveringInterval lists the operation logs that cover the interval
+func GetLogsCoveringInterval(folder storage.Folder, start time.Time, end *time.Time) ([]storage.Object, error) {
+	logFiles, _, err := folder.ListFolder()
 	if err != nil {
 		return nil, err
 	}
 
-	sort.Slice(oplogFiles, func(i, j int) bool {
-		return oplogFiles[i].GetLastModified().After(oplogFiles[j].GetLastModified())
-	})
-
 	var logsToFetch []storage.Object
 
-	for _, oplogFile := range oplogFiles {
-		if oplogFile.GetLastModified().After(start) {
-			logsToFetch = append(logsToFetch, oplogFile)
-			if end != nil && oplogFile.GetLastModified().After(*end) {
-				break
-			}
+	for _, logFile := range logFiles {
+		tracelog.InfoLogger.Println("Consider log file: ", logFile.GetName(), logFile.GetLastModified().Format(time.RFC3339))
+		if LogFileShouldBeFetched(start, end, logFile) {
+			logsToFetch = append(logsToFetch, logFile)
 		}
 	}
-	return logsToFetch, err
+	sort.Slice(logsToFetch, func(i, j int) bool {
+		return logsToFetch[i].GetLastModified().Before(logsToFetch[j].GetLastModified())
+	})
+	return logsToFetch, nil
 }
 
-// DownloadOplogFiles downloads files to specified folder
-func DownloadOplogFiles(oplogFiles []storage.Object, oplogFolder storage.Folder, oplogDstFolder string, logFileName string) error {
-	for _, oplogFile := range oplogFiles {
-		oplogName := utility.TrimFileExtension(oplogFile.GetName())
-		oplogFileSubFolder := path.Join(oplogDstFolder, oplogName)
-		_, err := NewDiskDataFolder(oplogFileSubFolder)
+// DownloadLogFiles downloads files to specified folder
+func DownloadLogFiles(logFiles []storage.Object, logFolder storage.Folder, logDstFolder string, getLogFilePath func(string, string) (string, error)) error {
```

Comment:
Is it possible to simplify it :/ ?

Reply:
It is definitely possible. But it appears to be a complicated task, so I decided to leave it as it is.